pkg/proxy/backend/backend_conn_mgr.go

```diff
@@ -798,3 +828,98 @@ func (mgr *BackendConnManager) ConnInfo() []zap.Field {
 	fields = append(fields, zap.String("backend_addr", mgr.ServerAddr()))
 	return fields
 }
+
+// Pause implements RedirectableConn.Pause interface. It notifies manager to pause.
+func (mgr *BackendConnManager) Pause() bool {
+	if mgr.closeStatus.Load() >= statusNotifyClose {
+		return false
+	}
+
+	if mgr.backendIO.Load() == nil {
+		return false
+	}
+
+	mgr.signalReceived <- signalTypePause
```

Comment:
It may block here if the previous signal is not processed yet. Thus, the whole pause procedure in the `Router` may block.

Reply:
If `Router` blocks here, `BackendObserver` can be blocked too. Then backend list can not be flushed correctly. Maybe it's better to do perform the pause procedure in another goroutine. Let me take a look carefully.

Reply:
Oh, I was wrong. It won't block here because the channel size is big enough and the same signal won't be sent again before it's processed.
Never mind.